src/PrestaShopBundle/Bridge/AdminController/AdminControllerTrait.php

```diff
@@ -97,45 +65,68 @@ public function addAction(ActionInterface $action): void
     }
 
     /**
-     * This method add action specific for the list.
+     * @param Request $request
      *
-     * @param ActionInterface $action
-     * @param HelperListConfiguration $helperListConfiguration
-     *
-     * @return void
+     * @return LegacyControllerBridgeInterface
      */
-    public function addActionList(ActionInterface $action, HelperListConfiguration $helperListConfiguration): void
+    public function initLegacyControllerBridge(Request $request): LegacyControllerBridgeInterface
```

Comment:
The same principle is implementable for the controller configuration actually since it's the one that carries all the config and can be heavily reused in the actually controller actions:

```php
    public function getControllerConfiguration(): ControllerConfiguration
    {
        return $this->buildControllerConfiguration('AdminFeatures', 'ps_features', Feature::class);
    }

    // And the build method for the bridge controller could probably look like this then
    public function getControllerBridge(): LegacyControllerBridgeInterface
    {
        return $this->buildControllerBridge($this->getControllerConfiguration());
    }
```

With such implementation we don't need the `getTable` and `getClassName` anymore, only two methods: `getControllerConfiguration` and `getControllerBridge` which seem to reflect more the actual responsibility of such a controller.

Reply:
>I don't think we should require the request on this method, the controller is supposed to know its own associated legacy controller

hm right, but we still need to provide _legacy_controller in route as we use it for security annotations, so isn't that abit redundant to also require additional method to be implemented for it?